sigstore-java/src/test/java/dev/sigstore/tuf/TufClientTest.java

```diff
@@ -146,6 +148,97 @@ public void testRootUpdate_metaFileTooBig()
     }
   }
 
+  @Test
+  public void testTimestampUpdate_throwMetaNotFoundException() throws IOException {
+    setupMirror("remote-timestamp-not-present");
+    var client = createTimeStaticTufClient(localStore);
+    assertThrows(
+        MetaNotFoundException.class,
+        () -> {
+          client.updateTimestamp();
+        });
+  }
+
+  @Test
+  public void testTimestampUpdate_throwsSignatureVerificationException()
+      throws IOException, NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException {
+    setupMirror("remote-timestamp-unsigned", "2.root.json", "3.root.json", "timestamp.json");
+    var client = createTimeStaticTufClient(localStore);
+    try {
+      client.updateRoot();
+      client.updateTimestamp();
+      fail();
+    } catch (SignatureVerificationException e) {
+      assertEquals(0, e.getVerifiedSignatures());
+      assertEquals(1, e.getRequiredSignatures());
```

Comment:
You might want to consider https://approvaltests.com/ for verifications like that. I do not insist on approvaltests, however, it might help us to mass-update the expected outcomes should the output format change.

Reply:
Wow that looks super interesting.  I don't want to learn and pull in a new testing framework for this case, but I'm very much open to taking a look at this and how it can improve these tests in the future.  If you'd like to send a demo PR so that we can see how it works I'd be 100% for it.